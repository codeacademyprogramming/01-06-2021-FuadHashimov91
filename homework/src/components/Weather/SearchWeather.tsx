const SearchWeather = ({ setCity, getData }: any) => {
    return (
        <div className='col-12'>
            <div className="input-group d-flex justify-content-center mt-4" >
                <div className="form-outline d-flex">
                    <label className='col'>Search</label>
                    <input
                        name="searchInput"
                        required
                        onChange={(e) => setCity(e.target.value)}
                        type="search" id="form1" className="form-control" placeholder="Search"
                    />
                </div>
                <button
                    test-id = 'searchBtn'
                    onClick={() => getData()}
                    type="button" className=" btn btn-primary">Search
                </button>
            </div>
        </div>
    )
}
export default SearchWeather;