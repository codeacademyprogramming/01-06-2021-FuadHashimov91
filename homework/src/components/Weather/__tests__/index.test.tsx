import React from 'react';
import renderer from "react-test-renderer";
import Weather from '../Weather';
import ReactDOM from 'react-dom';

describe('Weather Component', () => {
    it('the snapshot', () => {
        const tree = renderer.create(<Weather />).toJSON();
        expect(tree).toMatchSnapshot()
    });

    test("Test Header", () => {
        const root = document.createElement("div");
        ReactDOM.render
        (
            <Weather />,
            root
        );
        expect(
            root.querySelector("#head")?.textContent
        ).toBe('Weather App')
    })
})




