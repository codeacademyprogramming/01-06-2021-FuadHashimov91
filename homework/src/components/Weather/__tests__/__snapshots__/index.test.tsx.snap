// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Component the snapshot 1`] = `
Array [
  <h1
    className="text-center m-5"
    id="head"
  >
    Weather App
  </h1>,
  <div
    className="col-12"
  >
    <div
      className="input-group d-flex justify-content-center mt-4"
    >
      <div
        className="form-outline d-flex"
      >
        <label
          className="col"
        >
          Search
        </label>
        <input
          className="form-control"
          id="form1"
          onChange={[Function]}
          placeholder="Search"
          required={true}
          type="search"
        />
      </div>
      <button
        className=" btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="App row"
    >
      <div
        className="col-12 card mt-5"
      >
        <div />
      </div>
    </div>
  </div>,
]
`;
